cmd/gitserver/server/cleanup.go

```diff
@@ -194,44 +194,75 @@ func (s *Server) cleanupRepos() {
 		log15.Error("cleanup: error iterating over repositories", "error", err)
 	}
 
-	// Check how much disk space is available.
-	actualFreeBytes, err := s.bytesFreeOnDisk()
-	if err != nil {
-		log15.Error("cleanup: finding the amount of space free on disk", "error", err)
-		return
+	if s.DiskSizer == nil {
+		s.DiskSizer = &StatDiskSizer{}
 	}
-	G := float64(1024 * 1024 * 1024)
-	mount, err := findMountPoint(s.ReposDir)
+	btf, err := s.howManyBytesToFree()
 	if err != nil {
-		log15.Error("finding mount point for dir containing repos", "error", err)
-		mount = "<not found>"
+		log15.Error("cleanup: ensuring free disk space", "error", err)
 	}
-	log15.Debug("cleanup",
-		"free space in GiB", float64(actualFreeBytes)/G,
-		"desired free space in GiB", float64(s.DesiredFreeDiskSpace)/G,
-		"mount point", mount)
-
-	// Free up space if necessary.
-	howManyBytesToFree := int64(s.DesiredFreeDiskSpace) - int64(actualFreeBytes)
-	if err := s.freeUpSpace(howManyBytesToFree); err != nil {
+	if err := s.freeUpSpace(btf); err != nil {
 		log15.Error("cleanup: error freeing up space", "error", err)
 	}
 }
 
-// bytesFreeOnDisk tells how much space is available on the disk containing s.ReposDir.
-func (s *Server) bytesFreeOnDisk() (uint64, error) {
-	var fs syscall.Statfs_t
-	mount, err := findMountPoint(s.ReposDir)
+// DiskSizer gets information about disk size and free space.
+type DiskSizer interface {
+	BytesFreeOnDisk(mountPoint string) (uint64, error)
+	DiskSizeBytes(mountPoint string) (uint64, error)
+}
+
+// howManyBytesToFree returns the number of bytes that should be freed to make sure
+// there is sufficient disk space free to satisfy s.DesiredPercentFree.
+func (s *Server) howManyBytesToFree() (int64, error) {
+	// Check how much disk space is available.
+	mountPoint, err := findMountPoint(s.ReposDir)
 	if err != nil {
 		return 0, errors.Wrap(err, "finding mount point for dir containing repos")
 	}
-	if err := syscall.Statfs(mount, &fs); err != nil {
+	actualFreeBytes, err := s.DiskSizer.BytesFreeOnDisk(mountPoint)
+	if err != nil {
+		return 0, errors.Wrap(err, "cleanup: finding the amount of space free on disk")
+	}
+	// Free up space if necessary.
```

Comment:
```suggestion

	// Free up space if necessary.
```

nit: I made similar comments elsewhere in this PR, but I find this code to be a bit dense to read since it lacks newlines to separate logically distinct sections. 

Could you read through the advice in https://github.com/sourcegraph/sourcegraph/blob/master/doc/dev/code_style_guide.md#use-new-lines-intentionally and adapt your code accordingly? (Example: newlines after early `return` statements)

Reply:
Done.

Reply:
I'm less sure about adding newlines after early return statements. Happy to chat about it if it's affecting the readability.